Framework/Kernel/src/Matrix.cpp

```diff
@@ -1413,14 +1414,11 @@ Attempt to diagonalise the matrix IF symmetric
         else if (fabs(A.m_rawData[ip][iq]) > tresh) {
           double tanAngle, cosAngle, sinAngle;
           double h = Diag[iq] - Diag[ip];
-          if (static_cast<float>((fabs(h) + g)) == static_cast<float>(fabs(h)))
```

Comment:
Nice catch with this logic here.
Are there any additional tests in `MatrixTest` we can add to try and prevent this from regressing in the future?

Reply:
I've added an extra unit test to cover the case where one or both of the matrices has NaN values. This exposed the fact that my original change to Matrix::equals didn't work either (!) so have changed that also. Had previously been trying hard to avoid using std::isnan because it's a template function and no overload of std::isnan exists for T=int. I now cast the values to doubles so I can use std::isnan so bit more explicit\clear now hopefully

Reply:
Ah yes the template would complicate things. It's a shame that std::isnan<int> isn't there and just always returns false...

Thanks for adding the test.